src/predictor/gpu_predictor.cu

```diff
@@ -324,22 +344,27 @@ class GPUPredictor : public xgboost::Predictor {
                 h_nodes.begin() + h_tree_segments[tree_idx - tree_begin]);
     }
 
-    size_t i_batch = 0;
     size_t batch_offset = 0;
     for (auto &batch : dmat->GetRowBatches()) {
-      CHECK(i_batch == 0 || devices_.Size() == 1) << "External memory not supported for multi-GPU";
-      // out_preds have been sharded and resized in InitOutPredictions()
+      std::vector<size_t> out_preds_offsets;
+      PredictionDeviceOffsets(out_preds->Size(), batch_offset, batch.Size(),
+                              model.param.num_output_group, &out_preds_offsets);
+      out_preds->Reshard(GPUDistribution::Explicit(devices_, out_preds_offsets));
```

Comment:
1. Resharding will also copy back to host predictions from previous batches. However, this should be fine if your performance measurements don't show any problems.

2. A distribution with offsets is not equal to an implicit distribution with the same offsets. Please add avoiding extra copying in this situation in `Shard()` and `Reshard()`. No additional copying should happen when external memory is not used.

Reply:
For 1, I think the copying needs to happen since we are sliding over the predictions on the GPUs.

For 2, added a guard to avoid unnecessary resharding.